tiledb/sm/array_schema/attribute.h

```diff
@@ -107,9 +127,10 @@ class Attribute {
    *
    * @param buff The buffer to deserialize from.
    * @param version The format spec version.
-   * @return Status
+   * @return Status and Attribute
    */
-  Status deserialize(ConstBuffer* buff, uint32_t version);
+  static std::tuple<Status, std::optional<Attribute*>> deserialize(
```

Comment:
I need to work out how we can avoid an allocation in the factory. I don't want to rely on that before this PR is done. Therefore, to make progress we can use an allocation. But we don't want a raw allocation here; it's not exception-safe. Return value should have `tiledb_unique_ptr`. It'll require changing the corresponding type in the array schema class.

Even if we're not eliminating the allocation in this PR, it still makes progress toward C.41 compliance for this class.

Reply:
It is better to use raw pointer in this PR, since raw pointer of Attribute is stored in other classes, such as ArraySchema. I tried to create tiledb_shared_ptr or tiledb_unique_ptr here, but it causes a lot of problems. In future, when we change all raw pointer to smart pointer, we can change them together